cookbooks/aws-parallelcluster-scheduler-plugin/files/default/clusterstatusmgtd/clusterstatusmgtd.py

```diff
@@ -0,0 +1,450 @@
+        if proxy != "NONE":
+            self._boto3_config["proxies"] = {"https": proxy}
+        self.boto3_config = Config(**self._boto3_config)
+
+
+class ClusterStatusManager:
+    """The cluster status manager."""
+
+    def __init__(self, config):
+        """Initialize ClusterStatusManager."""
+        self._config = None
+        self._current_time = None
+        self._compute_fleet_status_manager = None
+        self._compute_fleet_status = ComputeFleetStatus.RUNNING
+        self._compute_fleet_data = {}
+
+    def set_config(self, config):  # noqa: D102
+        if self._config != config:
+            log.info(f"Applying new clusterstatusmgtd config: {config}")
+            self._config = config
+            self._compute_fleet_status_manager = self._initialize_compute_fleet_status_manager(config)
+
+    @staticmethod
+    def _initialize_compute_fleet_status_manager(config):
+        return ComputeFleetStatusManager(
+            table_name=config.dynamodb_table, boto3_config=config.boto3_config, region=config.region
+        )
+
+    def _get_compute_fleet_status(self, fallback=None):
+        try:
+            log.info("Getting compute fleet status")
+            self._compute_fleet_data = self._compute_fleet_status_manager.get_status()
+
+            return ComputeFleetStatus(
+                self._compute_fleet_data.get(self._compute_fleet_status_manager.COMPUTE_FLEET_STATUS_ATTRIBUTE)
+            )
+        except Exception as e:
+            log.error(
+                f"Failed when retrieving computefleet status from DynamoDB with error {e}, "
+                f"using fallback value {fallback}"
+            )
+            return fallback
+
+    def _update_compute_fleet_status(self, new_status):
+        log.info(f"Updating compute fleet status from {self._compute_fleet_status} to {new_status}")
+        self._compute_fleet_data = self._compute_fleet_status_manager.update_status(
+            current_status=self._compute_fleet_status, next_status=new_status
+        )
+        self._compute_fleet_status = new_status
+
+    def _call_update_event(self):
+        compute_fleet_data = ComputeFleetStatus._transform_compute_fleet_data(self._compute_fleet_data)
+        _write_json_to_file(self._config.computefleet_status_path, compute_fleet_data)
+
+        log.info("Calling update event handler, log can be found at /var/log/chef-client.log")
+        cinc_log_file = "/var/log/chef-client.log"
+        cmd = (
+            "sudo cinc-client "
+            "--local-mode "
+            "--config /etc/chef/client.rb "
+            "--log_level auto "
+            f"--logfile {cinc_log_file} "
+            "--force-formatter "
+            "--no-color "
+            "--chef-zero-port 8889 "
+            "--json-attributes /etc/chef/dna.json "
+            "--override-runlist aws-parallelcluster::update_computefleet_status"
+        )
+        _run_command(cmd, log_file=cinc_log_file)
```

Comment:
I think we should have a timeout here (15mins?) and make sure we document it somewhere. we could also make it configurable in the daemon config but it is not a hard requirement.